nextwind/models.py

```diff
@@ -14,81 +13,213 @@ def __init__(self, window):
+    
+    # Reshape (batch, sequences, features)
+    outputs = tf.keras.layers.Reshape(window.train['y'].shape[1:])(outputs)
+    
+    model = tf.keras.models.Model(inputs=[input_perf, input_fc], outputs=outputs)
+    
+    return model
+
+def gru_regressor_model(window):
+    # Performance model
+    input_perf = tf.keras.layers.Input(shape=window.train['X'].shape[1:])
+    x_seq = tf.keras.layers.LSTM(64, return_sequences=True, dropout=0.2)(input_perf)
+    x_seq = tf.keras.layers.LSTM(64, return_sequences=True, dropout=0.2)(x_seq)
+    x_perf = tf.keras.layers.LSTM(64, return_sequences=False, dropout=0.2)(x_seq)
+    
+    # Weather forecast model
+    input_fc = tf.keras.layers.Input(shape=window.train['X_fc'].shape[1:])
+    x_fc_seq = tf.keras.layers.LSTM(64, return_sequences=True, dropout=0.2)(input_fc)
+    x_fc_seq = tf.keras.layers.LSTM(64, return_sequences=True, dropout=0.2)(x_fc_seq)
+    x_fc = tf.keras.layers.LSTM(64, return_sequences=False, dropout=0.2)(x_fc_seq)
+    
+    # Combined model
+    combined = tf.keras.layers.concatenate([x_perf, x_fc], axis=1)
+    x = tf.keras.layers.Dense(128, activation='relu')(combined)
+    x = tf.keras.layers.Dense(64, activation='relu')(x)
+    outputs = tf.keras.layers.Dense(window.train['y'].shape[1], activation="relu")(x)
+    
+    # Reshape (batch, sequences, features)
+    outputs = tf.keras.layers.Reshape(window.train['y'].shape[1:])(outputs)
+    
+    model = tf.keras.models.Model(inputs=[input_perf, input_fc], outputs=outputs)
+    
+    return model
+
+
+def gru_classifier_model(window):
+    # Performance model
+    input_perf = tf.keras.layers.Input(shape=window.train['X'].shape[1:])
+    x_seq = tf.keras.layers.LSTM(64, return_sequences=True, dropout=0.2)(input_perf)
+    x_seq = tf.keras.layers.LSTM(64, return_sequences=True, dropout=0.2)(x_seq)
+    x_perf = tf.keras.layers.LSTM(64, return_sequences=False, dropout=0.2)(x_seq)
+    
+    # Weather forecast model
+    input_fc = tf.keras.layers.Input(shape=window.train['X_fc'].shape[1:])
+    x_fc_seq = tf.keras.layers.LSTM(64, return_sequences=True, dropout=0.2)(input_fc)
+    x_fc_seq = tf.keras.layers.LSTM(64, return_sequences=True, dropout=0.2)(x_fc_seq)
+    x_fc = tf.keras.layers.LSTM(64, return_sequences=False, dropout=0.2)(x_fc_seq)
+    
+    # Combined model
+    combined = tf.keras.layers.concatenate([x_perf, x_fc], axis=1)
+    x = tf.keras.layers.Dense(128, activation='relu')(combined)
+    x = tf.keras.layers.Dropout(0.2)(x)
+    x = tf.keras.layers.Dense(64, activation='relu')(x)
+    x = tf.keras.layers.Dropout(0.5)(x)
+    outputs = tf.keras.layers.Dense(3, activation="softmax")(x)
+    
+    # Reshape (batch, sequences, features)
+    outputs = tf.keras.layers.Reshape(window.train['y'].shape[1:])(outputs)
+    
+    model = tf.keras.models.Model(inputs=[input_perf, input_fc], outputs=outputs)
+    
+    return model
+
 
 
 class LSTM_Regressor_Model(tf.keras.Model):
```

Comment:
Currently does not function - to make performance and forecast models as separate classes, instead of inner classes